vault/quotas/quotas_rate_limit_test.go

```diff
@@ -146,14 +146,19 @@ func TestRateLimitQuota_Allow_WithBlock(t *testing.T) {
 
 	var wg sync.WaitGroup
 
-	reqFunc := func(addr string, atomicNumAllow, atomicNumFail *atomic.Int32) {
+	reqFunc := func(addr string, atomicNumAllow, atomicNumFail *atomic.Int32, start time.Time) {
 		defer wg.Done()
 
 		resp, err := rlq.allow(context.Background(), &Request{ClientAddress: addr})
 		if err != nil {
 			return
 		}
 
+		// Avoid slow CI passing the BlockInterval of the rate limiter.
+		if time.Now().After(start.Add(rlq.BlockInterval)) {
```

Comment:
I think this condition will never be met.

`start.Add(rlq.BlockInterval)` will be "now+10seconds" whereas the for loop which triggers this function runs for "now+5seconds".

This also makes me wonder if the test fix in this PR is passing due to sheer chance?

Reply:
Please let me know if I misread the fix.